syntax = "proto3";

package proto;

service StrategyManager {
  rpc SendSignal (SignalMessage) returns (ExecutionResponse);
}

message SignalMessage {
  string strategy_name = 1;
  string symbol = 2;
  string timeframe = 3;
  string action = 4;
  double confidence = 5;
  double entry_price = 6;
  double stop_loss = 7;
  double take_profit = 8;
  int64 timestamp = 9;
}

message ExecutionResponse {
  string status = 1;
  double score = 2;
  double capital = 3;
  string message = 4;
}
